### ISSUE TITLE
[BUG] [timeseries] DirectTabularModel fails 

### ISSUE THREAD

**Issue by noamanemobidata** [autogluon/autogluon#3341]
**Bug Report Checklist**

- [ ] I provided code that demonstrates a minimal reproducible example.  
- [ ] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [ ] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**
Training DirectTabularModel fails : 

Warning: Exception caused DirectTabular to fail during training... Skipping this model.
	'Series' object has no attribute 'set_index'
Traceback (most recent call last):
  File "/local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages/autogluon/timeseries/trainer/abstract_trainer.py", line 495, in _train_and_save
    model = self._train_single(train_data, model, val_data=val_data, time_limit=time_limit)
  File "/local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages/autogluon/timeseries/trainer/abstract_trainer.py", line 408, in _train_single
    model.fit(
  File "/local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages/autogluon/timeseries/models/abstract/abstract_timeseries_model.py", line 235, in fit
    return super().fit(**kwargs)
  File "/local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages/autogluon/core/models/abstract/abstract_model.py", line 811, in fit
    out = self._fit(**kwargs)
  File "/local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages/autogluon/timeseries/models/multi_window/multi_window_model.py", line 105, in _fit
    model.score_and_cache_oof(val_fold, store_val_score=True, store_predict_time=True)
  File "/local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages/autogluon/timeseries/models/abstract/abstract_timeseries_model.py", line 374, in score_and_cache_oof
    self._oof_predictions = self.predict(past_data, known_covariates=known_covariates)
  File "/local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages/autogluon/timeseries/models/autogluon_tabular/direct_tabular.py", line 389, in predict
    predictions.set_index(data_future.index, inplace=True)
  File "/databricks/python/lib/python3.10/site-packages/pandas/core/generic.py", line 5575, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'Series' object has no attribute 'set_index'
 
**Expected behavior**

**To Reproduce**

```python
from autogluon.timeseries.splitter import MultiWindowSplitter
splitter = MultiWindowSplitter(num_windows=1)
h=24
predictor = TimeSeriesPredictor(
  path="ts3/", 
  prediction_length=24,
  target="y",
  eval_metric="RMSE", 
  known_covariates_names=["active","age","OUT","localName", "close", "covid"],
  quantile_levels=[0.5], 
  #validation_splitter=MultiWindowSplitter(num_windows=1), 
  verbosity=4
).fit(
  train, 
  hyperparameters={
    "DirectTabularModel": {"tabular_hyperparameters": {"GBM":{} }}, 
    "RecursiveTabularModel": {"tabular_hyperparameters": {"GBM":{} }}
    }, 
  time_limit=3600*2
)

```

**Screenshots / Logs**

**Installed Versions**

<details>

```python
INSTALLED VERSIONS
------------------
date                   : 2023-06-21
time                   : 14:37:10.854780
python                 : 3.10.6.final.0
OS                     : Linux
OS-release             : 5.15.0-1038-azure
Version                : #45~20.04.1-Ubuntu SMP Tue Apr 25 18:45:15 UTC 2023
machine                : x86_64
processor              : x86_64
num_cores              : 32
cpu_ram_mb             : 249480
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : []
avail_disk_size_mb     : 228130

accelerate             : 0.16.0
autogluon              : 0.8.0
autogluon.common       : 0.8.0
autogluon.core         : 0.8.0
autogluon.features     : 0.8.0
autogluon.multimodal   : 0.8.0
autogluon.tabular      : 0.8.0
autogluon.timeseries   : 0.8.0
boto3                  : 1.24.28
catboost               : 1.2
defusedxml             : 0.7.1
evaluate               : 0.3.0
fairscale              : 0.4.13
fastai                 : 2.7.12
gluonts                : 0.13.2
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.2
joblib                 : 1.2.0
jsonschema             : 4.16.0
lightgbm               : 3.3.5
matplotlib             : 3.5.2
mlforecast             : 0.7.3
networkx               : 3.1
nlpaug                 : 1.1.11
nltk                   : 3.7
nptyping               : 2.4.1
numpy                  : 1.24.3
omegaconf              : 2.2.3
onnxruntime-gpu        : None
openmim                : 0.3.7
pandas                 : 1.4.4
Pillow                 : 9.5.0
psutil                 : 5.9.0
PyMuPDF                : 1.21.1
pytesseract            : 0.3.10
pytorch-lightning      : 1.9.5
pytorch-metric-learning: 1.7.3
ray                    : 2.3.1
requests               : 2.28.1
scikit-image           : 0.19.3
scikit-learn           : 1.1.1
scikit-learn-intelex   : None
scipy                  : 1.9.1
sentencepiece          : 0.1.99
seqeval                : 1.2.2
setuptools             : 63.4.1
skl2onnx               : None
statsforecast          : 1.4.0
statsmodels            : 0.13.2
tabpfn                 : None
tensorboard            : 2.11.0
text-unidecode         : 1.3
timm                   : 0.9.2
torch                  : 1.13.1+cpu
torchmetrics           : 0.11.4
torchvision            : 0.14.1+cpu
tqdm                   : 4.64.1
transformers           : 4.26.1
ujson                  : 5.4.0
vowpalwabbit           : None
xgboost                : 1.7.4```

</details>

**Comment by noamanemobidata** [comment]
the bug persists even when changing the settings  quantile_levels=[0.1, 0.5]

**Comment by shchur** [comment]
What pandas version are you using?

**Comment by shchur** [comment]
Could you please share the dataset that you're training on?

I tried creating a conda environment with the specification that you provided & running your code on several datasets, but wasn't able to reproduce the issue so far.

**Comment by shchur** [comment]
The problem is now fixed in AutoGluon v0.8.1 that is available for installation via `pip`.
